src/base/name/builder.rs

```diff
@@ -543,51 +538,54 @@ impl std::error::Error for PushNameError {}
 
 /// An error occured while reading a label from a string.
 #[derive(Clone, Copy, Debug, Eq, PartialEq)]
-pub enum LabelFromStrError {
-    /// The string ended when there should have been more characters.
-    ///
-    /// This most likely happens inside escape sequences and quoting.
-    UnexpectedEnd,
+pub struct LabelFromStrError(LabelFromStrErrorEnum);
+
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub(super) enum LabelFromStrErrorEnum {
+    SymbolChars(SymbolCharsError),
```

Comment:
Why does this inner type have an `Error` suffix but the `BadSymbol` one does not?

Reply:
I just figured it doesn’t need the suffix because the name kind of makes it obvious that it is an error type – and a shorter name is nicer?